### DISCUSSION TITLE
Curious about Lab5 step 4

### DISCUSSION THREAD

**Discussion by Robert303V** [hack4impact-calpoly/starter-pack#4] [Q&A] [answered]
This isn't really a question I need help with, more a bit of curiosity. On Lab5 (setup Git) step 4 says 'Setup Line Ending Preferences'. I've worked with git before but am not familiar with Line Ending Preferences, what exactly do these commands do?

**Comment by TeachMeTW** [comment, marked as answer]
It helps collaborators from different operating systems be more "in sync" with each other. For example, Windows uses '\r\n' for new line endings while Mac uses only '\n'. Sometimes this causes unwarranted changes/discrepancies. This is where git solves this issue by converting the line ending format to what is being used/compatible.

This is how I interpreted it at least.

**Reply by khoa-l** [reply]
Yup! Perfect explanation. We have developers on both Mac and Windows, so this just helps solve any issues that could possibly arise from conflicting line endings.

**Reply by Robert303V** [reply]
Oh ok, thanks for the explanation!
